src/table/virtualized-table/Body.tsx

```diff
@@ -77,29 +73,31 @@ const Body = forwardRef<BodyRef, BodyProps>((props, ref) => {
     [rowsInPage, columns, bodyStyle, isHoverEnabled, maxLineCount]
   );
 
-  const bodyHeight = getBodyHeight(rect, headerAndTotalsHeight, deferredRowCount, estimatedRowHeight);
+  const bodyHeight = getBodyHeight(rect, headerAndTotalsHeight, rowCount, estimatedRowHeight);
 
-  useEffect(() => {
+  useLayoutEffect(() => {
```

Comment:
is below `useLayoutEffect` trigger, not syncing height on every render? if so, do we still need this one too?

Reply:
That is correct and intended. The first `useLayoutEffect` (with deps) tells `syncHeigt` to forcefully run no matter what. Sort like a `forceUpdate` call.

The second `useLayoutEffect` is using a guard inside that syncHeight function, so that in most renders, it does not actually do anything. Only given some condition is meet does it update a state.